tinygrad/codegen/uopgraph.py

```diff
@@ -205,14 +207,18 @@ def index_collapse(idx,rng,buf,add,mul,ld,reduce_allow_any_len):
   (NOp.max(NOp.cvar('c'), -(NOp(UOps.RANGE, name="s")+NOp.cvar('c2'))), lambda c,s,c2: -(s+c2) if -(s.src[1].arg-1+c2.arg) >= c.arg else None),
   # const rules
   (NOp(UOps.GEP, src=(NOp.cvar("c"),), name="root"), lambda root, c: root.const(c.arg)),
+  *[(UPat(UOps.GEP, name="root", src=(UPat(UOps.VECTORIZE, name="v", src=[UPat(UOps.CONST) for _ in range(i)]))),
+     lambda root, v: root.const(v.src[root.arg].arg)) for i in [2, 4, 8]],
```

Comment:
isn't v.src[root.arg] const already?

Reply:
yes. I simplified the logic. I saw this in two other unrelated pats as well. Will push a separate PR for that.

Reply:
cool, those PRs need [run_process_replay] as well.

Reply:
will do. that change is actually dependent on this one, so I'll push and tag later.